src/Shared/ListHelper.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Sven\ArtisanView\Shared;
+
+class ListHelper
+{
+    /**
+    * @param $directory
+    *
+    * @return string Tree view as single string.
+    */
+   public function getListAsString($directory, $level = 0)
+   {
+       if (!file_exists($directory)) {
+           return false;
+       }
+
+       $result = '';
+
+       if ($this->directoryContainsViews($directory)) {
+           $items = array_diff(scandir($directory), ['.', '..']);
+           foreach ($items as $item) {
+               if ($this->directoryContainsViews($directory.'/'.$item)) {
+                   $result .= PHP_EOL.$this->getIndentation($level).$item;
+                   if (is_dir($directory.'/'.$item)) {
+                       $result .= $this->getListAsString($directory.'/'.$item, $level + 1);
```

Comment:
I can't think of a way to clean this function. I guess the ugly part is having multiple nested if-statements inside the for-loop. However, there needs to be a check before the next level of directories & views gets appended to the result. Any ideas?